js-core/homeworks/phoneApp/src/main.js

```diff
@@ -0,0 +1,111 @@
+function PhoneApp() {
+    this.storage = [];
+}
+
+PhoneApp.prototype.checkNumber = function(numberToCheck) {
+    const toStringThePredictNumber = numberToCheck.toString();
+    const toArrayTheString = toStringThePredictNumber.split('');
+    let flag;
+    
+    toArrayTheString.forEach(elem => {
+        const toNumberElem = Number(elem);
+        if(isNaN(toNumberElem)) {
+            flag = false;
+        }
+    });
+    
+    if(flag === false) {
+        return flag;
+    } else {
+        return true;
+    }
+}
+
+PhoneApp.prototype.sortNumber = function(numberToSort) {
+    if(this.checkNumber(numberToSort)) {
+        const toStringTheNumber = numberToSort.toString();
+        const firstThreeNumbers = toStringTheNumber.slice(0, 3);
+        const bracketThisFirstThreeNumbers = `(${firstThreeNumbers}) `;
+        
+        const secondTwoNumbers = toStringTheNumber.slice(3, 5);
+        const bracketThisSecondTwoNumbers = `${secondTwoNumbers}-`;
+        
+        const thirdTwoNumbers = toStringTheNumber.slice(5, 7);
+        const bracketThisThirdTwoNumbers = `${thirdTwoNumbers}-`;
+        
+        let lastNumbers = '';
+        for(let i = 7; i < toStringTheNumber.length; i++) {
```

Comment:
What the motivation to use a loop here? can we avoid it ?

Reply:
I mean No
Explain it on classroom